insights/parsers/tests/test_samba_smb_conf.py

```diff
@@ -0,0 +1,245 @@
+#
+# You can choose a non default printing system using the Printing option
+
+    load printers = yes
+    cups options = raw
+
+;   printcap name = /etc/printcap
+    #obtain list of printers automatically on SystemV
+;   printcap name = lpstat
+;   printing = cups
+
+# --------------------------- Filesystem Options ---------------------------
+#
+# The following options can be uncommented if the filesystem supports
+# Extended Attributes and they are enabled (usually by the mount option
+# user_xattr). Thess options will let the admin store the DOS attributes
+# in an EA and make samba not mess with the permission bits.
+#
+# Note: these options can also be set just per share, setting them in global
+# makes them the default for all shares
+
+;   map archive = no
+;   map hidden = no
+;   map read only = no
+;   map system = no
+;   store dos attributes = yes
+
+
+#============================ Share Definitions ==============================
+
+[homes]
+    comment = Home Directories
+    browseable = no
+    writable = yes
+;   valid users = %S
+;   valid users = MYDOMAIN\%S
+
+[printers]
+    comment = All Printers
+    path = /var/spool/samba
+    browseable = no
+    guest ok = no
+    writable = no
+    printable = yes
+
+# Un-comment the following and create the netlogon directory for Domain Logons
+;   [netlogon]
+;   comment = Network Logon Service
+;   path = /var/lib/samba/netlogon
+;   guest ok = yes
+;   writable = no
+;   share modes = no
+
+
+# Un-comment the following to provide a specific roving profile share
+# the default is to use the user's home directory
+;   [Profiles]
+;   path = /var/lib/samba/profiles
+;   browseable = no
+;   guest ok = yes
+
+
+# A publicly accessible directory, but read only, except for people in
+# the "staff" group
+;   [public]
+;   comment = Public Stuff
+;   path = /home/samba
+;   public = yes
+;   writable = yes
+;   printable = no
+;   write list = +staff
+
+[ GlObAl  ]
+
+# Samba also automatically treats non-lowercase section names as lowercase and strips whitespace.
+# This behavior can be checked with `testparm` again.
+this option should also be in global = true
+
+[ GlObAl  ]
+
+# This tests specifically that two same-named sections are automatically merged in case the
+# RawConfigParser's behavior ever changes.
+this another option should also be in global = 1
+"""
+
+# This is the output of testparm -s parsing the above SAMBA_CONFIG
+SAMBA_TESTPARM_OUTPUT = '''
+# Global parameters
+[global]
+        log file = /var/log/samba/log.%m
+        max log size = 50
+        security = USER
+        server string = Samba Server Version %v
+        workgroup = MYGROUP
+        idmap config * : backend = tdb
+        cups options = raw
+
+
+[homes]
+        browseable = No
+        comment = Home Directories
+        read only = No
+
+
+[printers]
+        browseable = No
+        comment = All Printers
+        path = /var/spool/samba
+        printable = Yes
+'''
+
+
+def test_match():
```

Comment:
I see that you have the `allow_no_values` set for the parser, are you testing any keys with no values?

Reply:
We don't have it set, but I set it and added a test now.